tests/queries/models.py

```diff
@@ -748,3 +748,23 @@ class JSONFieldNullable(models.Model):
 
     class Meta:
         required_db_features = {'supports_json_field'}
+
+
+# Models for #33319 regression tests
+
+class Foo(models.Model):
+    qux = models.ForeignKey("Qux", on_delete=models.CASCADE, related_name="foos")
+
+
+class Bar(models.Model):
+    foo = models.ForeignKey("Foo", on_delete=models.CASCADE, related_name="bars")
+    another_foo = models.ForeignKey("Foo", on_delete=models.CASCADE, related_name="other_bars")
+    baz = models.ForeignKey("Baz", on_delete=models.CASCADE, related_name="bars")
+
+
+class Baz(models.Model):
+    pass
+
+
+class Qux(models.Model):
+    bazes = models.ManyToManyField("Baz", related_name="quxes")
```

Comment:
It'd be great we if we could avoid creating 5 new tables to reproduce the issue. Existing ones should be reusable somehow.